server/stats.go

```diff
@@ -49,6 +50,85 @@ const lowestRank = 100
 // Limit the concurrent channels when processing in-memory cache data.
 const maxChannelSize = 50
 
+// byRank implements sort.Interface for []*SourceSeries based on
+// the rank score.
+type byRank []*SourceSeries
+
+func (a byRank) Len() int {
+	return len(a)
+}
+func (a byRank) Swap(i, j int) {
+	a[i], a[j] = a[j], a[i]
+}
+func (a byRank) Less(i, j int) bool {
+	oi := a[i]
+	keyi := rankKey{oi.ImportName, oi.MeasurementMethod}
+	scorei, ok := statsRanking[keyi]
+	if !ok {
+		scorei = lowestRank
+	}
+	oj := a[j]
+	keyj := rankKey{oj.ImportName, oj.MeasurementMethod}
+	scorej, ok := statsRanking[keyj]
+	if !ok {
+		scorej = lowestRank
+	}
+	// Higher score value means lower rank.
+	return scorei < scorej
```

Comment:
For determinism, should we also include the other fields in sort decision: observationPeriod, unit, scalingFactor.

Reply:
SG. Added a TODO for this. Will fix when this set of api work is done.